src/java/org/apache/cassandra/config/DataRateSpec.java

```diff
@@ -285,27 +301,12 @@ public IntMebibytesPerSecondBound(double quantity, DataRateUnit unit)
         /**
          * Creates a {@code DataRateSpec.IntMebibytesPerSecondBound} of the specified amount in mebibytes per second.
          *
-         * @param mebibytesPerSecond where mebibytesPerSecond shouldn't be bigger than Long.MAX_VALUE-1
+         * @param mebibytesPerSecond where mebibytesPerSecond shouldn't be bigger than Integer.MAX_VALUE-1
```

Comment:
nitty nit: We call this `max` in the downstream `validateQuantity()`, although it's actually one more than the max. I know I'm getting pedantic now, but perhaps we should just pass `Integer.MAX_VALUE - 1` and use `>` :D

Reply:
Correct, but let's not do it now, please... the patch became super big and If we change it here I think we have the same pattern in `DataStorageSpec` and `DurationSpec`...